### PR TITLE
New Basemaps: Local Server and Map View Categories

### PR DESCRIPTION
**Pull request by jenmerritt**
### Description

This PR is for the following sample categories:
- Local Server
- Map View

### General comments on Basemap updates:

It updates `Map` and/or `Basemap` Constructors as per the new design. This may include the following changes depending on the sample:
- Replace `Basemap.Type` with `BasemapStyle`
- If a lat/long/zoom was being set in the `Map` constructor, this is replaced with a `Viewpoint` being set on the `MapView`.
- Any Readme Updates identified

In addition, the samples that include the above changes, or other services requiring an API Key going forward, have the API Key  set within the sample. This will then work in conjunction with the Gradle Task being added in PR 593.

Once 593 is approved and merged into `dev` we will merge those changes into these PRs and then update all the Samples to `100.10.0-xxxx` and do a round of testing of the Samples with all the API Key / Basemap changes at that time. (We have left them as 100.9.0 (in most cases) for now to minimize merge conflicts in the `build.gradle` file).

In the meantime, the updates listed above can be reviewed. Thanks!

### CODE REVIEW

File: map_view/display-drawing-status/src/main/java/com/esri/samples/display_drawing_status/DisplayDrawingStatusSample.java [hunk 1]
```diff
@@ -55,12 +56,16 @@ public void start(Stage stage) {
...
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
       // create progress bar
       ProgressBar progressBar = new ProgressBar();
       progressBar.setMaxWidth(240.0);
 
-      // create a ArcGISMap with topographic basemap
-      final ArcGISMap map = new ArcGISMap(Basemap.createTopographic());
+      // create an ArcGISMap with a topographic basemap style
```
Comment (JonLavi) [inline map_view/display-drawing-status/src/main/java/com/esri/samples/display_drawing_status/DisplayDrawingStatusSample.java]: Let's do plain english here as well:
```suggestion
      // create a map with the topographic basemap style
```

File: map_view/display-layer-view-state/src/main/java/com/esri/samples/display_layer_view_state/DisplayLayerViewStateSample.java [hunk 2]
```diff
@@ -70,8 +71,12 @@ public void start(Stage stage) {
       stage.setScene(scene);
       stage.show();
 
-      // create an ArcGISMap with the topographic basemap
-      ArcGISMap map = new ArcGISMap(Basemap.createTopographic());
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create an ArcGISMap with the topographic basemap style
```
Comment (JonLavi) [inline map_view/display-layer-view-state/src/main/java/com/esri/samples/display_layer_view_state/DisplayLayerViewStateSample.java]: Same here:
```suggestion
      // create a map with the topographic basemap style
```

File: map_view/display-layer-view-state/src/main/java/com/esri/samples/display_layer_view_state/DisplayLayerViewStateSample.java [hunk 3]
```diff
@@ -70,8 +71,12 @@ public void start(Stage stage) {
...
+      ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC);
 
       // create a map view and set the ArcGISMap to it
```
Comment (JonLavi) [inline map_view/display-layer-view-state/src/main/java/com/esri/samples/display_layer_view_state/DisplayLayerViewStateSample.java]: Sneaky `ArcGISMap ` here as well ( I won't comment if there's any other instances):

```suggestion
      // create a map view and set the map to it
```

File: map_view/identify-layers/src/main/java/com/esri/samples/identify_layers/IdentifyLayersSample.java [hunk 4]
```diff
@@ -73,14 +74,20 @@ public void start(Stage stage) {
...
+
+      // create a map with a basemap style
+      ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC);
 
       // set the map to a map view
       mapView = new MapView();
       mapView.setMap(map);
 
+      // set a viewpoint on the map view
+      mapView.setViewpoint(new Viewpoint(new Point(-10977012.785807, 4514257.550369, SpatialReference.create(3857)), 68015210));
```
Comment (JonLavi) [inline map_view/identify-layers/src/main/java/com/esri/samples/identify_layers/IdentifyLayersSample.java]: Could use `SpatialReferences.getWebMercator()` here as well?

File: map_view/show-location-history/src/main/java/com/esri/samples/show_location_history/ShowLocationHistorySample.java [hunk 5]
```diff
@@ -81,8 +82,12 @@ public void start(Stage stage) {
       stage.show();
       scene.getStylesheets().add(getClass().getResource("/show_location_history/style.css").toExternalForm());
 
-      // create a map with a dark gray canvas basemap
-      ArcGISMap map = new ArcGISMap(Basemap.createDarkGrayCanvasVector());
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create a map with a dark gray basemap style
```
Comment (JonLavi) [inline map_view/show-location-history/src/main/java/com/esri/samples/show_location_history/ShowLocationHistorySample.java]:
```suggestion
      // create a map with the dark gray basemap style
```

**Review by JonLavi — APPROVED** [review]
LGTM @jenmerritt , just a few small suggestions

File: map_view/identify-layers/src/main/java/com/esri/samples/identify_layers/IdentifyLayersSample.java (hunk 4 above)
Comment (jenmerritt) [reply map_view/identify-layers/src/main/java/com/esri/samples/identify_layers/IdentifyLayersSample.java]: Have removed the `Point` from this as per other changes.

File: map_view/display-drawing-status/src/main/java/com/esri/samples/display_drawing_status/DisplayDrawingStatusSample.java [hunk 6]
```diff
@@ -55,21 +56,25 @@ public void start(Stage stage) {
...
-      // create a starting viewpoint for the ArcGISMap
+      // create a starting viewpoint for the map
       SpatialReference spatialReference = SpatialReferences.getWebMercator();
       Point bottomLeftPoint = new Point(-1.5054808160556655E7, 2718975.702666207, spatialReference);
       Point topRightPoint = new Point(-6810317.90634398, 6850505.377826911, spatialReference);
       Envelope envelope = new Envelope(bottomLeftPoint, topRightPoint);
       Viewpoint viewpoint = new Viewpoint(envelope);
 
-      // set the initial viewpoint of ArcGISMap to viewpoint
+      // set the initial viewpoint of the map
```
Comment (Rachael-E) [inline map_view/display-drawing-status/src/main/java/com/esri/samples/display_drawing_status/DisplayDrawingStatusSample.java]: was there a reason not to use `mapView.setViewpoint` here?

**Review by Rachael-E — COMMENTED** [review]
just a query here @jenmerritt

File: map_view/display-drawing-status/src/main/java/com/esri/samples/display_drawing_status/DisplayDrawingStatusSample.java (hunk 6 above)
Comment (jenmerritt) [reply map_view/display-drawing-status/src/main/java/com/esri/samples/display_drawing_status/DisplayDrawingStatusSample.java]: @Rachael-E think it just got missed, thanks! I've amended this and slightly reduced the number of separate lines of constructors. See what you think?

**Review by Rachael-E — APPROVED** [review]

**Comment by jenmerritt** [comment]
`dev` merged in (including API Key gradle tasks changes). Samples that have an API Key set have been tested running from the IDE.

Await merging these changes until release.

**Pull request merged**